cmd/namespace_registry.go

```
@@ -0,0 +1,253 @@
+	if err != nil {
+		return nil, err
+	}
+	var jwks JWKS
+	err = json.Unmarshal(keyInBytes, &jwks)
+	if err != nil {
+		return nil, err
+	}
+	keyData := jwks.Keys[0]  // Assumes there's at least one key
+	xBytes, _ := base64.RawURLEncoding.DecodeString(keyData.X)
+	yBytes, _ := base64.RawURLEncoding.DecodeString(keyData.Y)
+	x := new(big.Int).SetBytes(xBytes)
+	y := new(big.Int).SetBytes(yBytes)
+	publicKey := &ecdsa.PublicKey{
+		Curve: elliptic.P521(),
+		X:     x,
+		Y:     y,
+	}
+	return publicKey, nil
+}
+
+
+func signPayload(payload []byte, privateKey *ecdsa.PrivateKey) ([]byte, error) {
+    hash := sha256.Sum256(payload)
+    signature, err := privateKey.Sign(rand.Reader, hash[:], crypto.SHA256)  // Use crypto.SHA256 instead of the hash[:]
+    if err != nil {
+        return nil, err
+    }
+    return signature, nil
+}
+
+/*
+func writeSignatureToFile(signature []byte, filename string) error {
+	err := ioutil.WriteFile(filename, signature, 0644)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func loadSignatureFromFile(filename string) ([]byte, error) {
+	signature, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return nil, err
+	}
+	return signature, nil
+}
+
+func verifySignature(payload []byte, signature []byte, publicKey *ecdsa.PublicKey) bool {
+	hash := sha256.Sum256(payload)
+	return ecdsa.VerifyASN1(publicKey, hash[:], signature)
+}
+*/
+
+func generateNonce() (string, error) {
+    nonce := make([]byte, 32)
+    _, err := rand.Read(nonce)
+    if err != nil {
+        return "", err
+    }
+    return hex.EncodeToString(nonce), nil
+}
+
+func make_request(url string, method string, data map[string]interface{}) ([]byte) {
+	payload, _ := json.Marshal(data)
+
+	req, err := http.NewRequest(method, url, bytes.NewBuffer(payload))
+	if err != nil {
+		panic(err)
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+
+	client := &http.Client{}
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
```

Avoid panic'ing within the CLI -- let's instead propagate the error upward through a return variable.

---

✅